src/dscanner/analysis/nolint.d

```diff
@@ -0,0 +1,202 @@
+			this.disabledChecks ~= other.get.getDisabledChecks();
+	}
+
+private:
+		string[] disabledChecks;
+}
+
+struct NoLintFactory
+{
+	static Nullable!NoLint fromModuleDeclaration(in ModuleDeclaration moduleDeclaration)
+	{
+		NoLint noLint;
+
+		foreach(atAttribute; moduleDeclaration.atAttributes)
+			noLint.merge(NoLintFactory.fromAtAttribute(atAttribute));
+
+		if(!noLint.getDisabledChecks.length)
+			return nullNoLint;
+
+		return noLint.nullable;
+	}
+
+	static Nullable!NoLint fromDeclaration(in Declaration declaration)
+	{
+		NoLint noLint;
+		foreach(attribute; declaration.attributes)
+			noLint.merge(NoLintFactory.fromAttribute(attribute));
+
+		if(!noLint.getDisabledChecks.length)
+			return nullNoLint;
+
+		return noLint.nullable;
+	}
+
+
+private:
+	static Nullable!NoLint fromAttribute(const(Attribute) attribute)
+	{
+		if(attribute is null)
+			return nullNoLint;
+
+		return NoLintFactory.fromAtAttribute(attribute.atAttribute);
+
+	}
+
+	static Nullable!NoLint fromAtAttribute(const(AtAttribute) atAttribute)
+	{
+		if(atAttribute is null)
+			return nullNoLint;
+
+		auto ident = atAttribute.identifier;
+		auto argumentList = atAttribute.argumentList;
+
+		if(argumentList !is null)
+		{
+			if(ident.text.length)
+				return NoLintFactory.fromStructUda(ident, argumentList);
+			else
+				return NoLintFactory.fromStringUda(argumentList);
+
+		}
+		else
+			return nullNoLint;
+	}
+
+	// @nolint("..")
+	static Nullable!NoLint fromStructUda(in Token ident, in ArgumentList argumentList)
+	in(ident.text.length && argumentList !is null)
+	{
+		if(ident.text != "nolint")
+			return nullNoLint;
+
+		NoLint noLint;
+
+		foreach(nodeExpr; argumentList.items)
+		{
+			if(auto unaryExpr = cast(UnaryExpression) nodeExpr)
+			{
+				auto primaryExpression = unaryExpr.primaryExpression;
+				if(primaryExpression is null)
+					continue;
+
+				if(primaryExpression.primary != tok!"stringLiteral")
+					continue;
+
+				noLint.addCheck(primaryExpression.primary.text.strip("\""));
+			}
+		}
+
+		if(!noLint.getDisabledChecks().length)
+			return nullNoLint;
+
+		return noLint.nullable;
+	}
+
+	// @("nolint(..)")
+	static Nullable!NoLint fromStringUda(in ArgumentList argumentList)
+	in(argumentList !is null)
+	{
+		NoLint noLint;
+
+		foreach(nodeExpr; argumentList.items)
+		{
+			if(auto unaryExpr = cast(UnaryExpression) nodeExpr)
+			{
+				auto primaryExpression = unaryExpr.primaryExpression;
+				if(primaryExpression is null)
+					continue;
+
+				if(primaryExpression.primary != tok!"stringLiteral")
+					continue;
+
+				auto str = primaryExpression.primary.text.strip("\"");
+				Nullable!NoLint currNoLint = NoLintFactory.fromString(str);
+				noLint.merge(currNoLint);
+			}
+		}
+
+		if(!noLint.getDisabledChecks().length)
+			return nullNoLint;
+
+		return noLint.nullable;
+
+	}
+
+	// Transform a string with form "nolint(abc, efg)"
+	// into a NoLint struct
+	static Nullable!NoLint fromString(in string str)
+	{
+		auto re = regex(`\w+`, "g");
```

Comment:
```suggestion
		auto re = regex(`[\w_.]+`, "g");
```

checks can contain underlines and also we probably want to filter by error key, not by linter type, since some linters emit different kinds of errors, where you might only want to exclude some. Also since it's split by `.` it would be quite easy to implement wildcards by simply using `startsWith`

Users would also be able to `@nolint("dscanner")` and since all keys start with `dscanner.`, we also leave open this UDA / string pattern for other linters in the ecosystem and can verify that dscanner keys are correctly written.